core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java

```diff
@@ -92,656 +61,7 @@ public CatalogOperationDispatcher(
-
-    // Currently we only support maintaining the Fileset in the Gravitino's store.
-    return EntityCombinedFileset.of(fileset)
-        .withHiddenPropertiesSet(
-            getHiddenPropertyNames(
-                catalogIdent,
-                HasPropertyMetadata::filesetPropertiesMetadata,
-                fileset.properties()));
-  }
-
-  @Override
-  public Fileset createFileset(
-      NameIdentifier ident,
-      String comment,
-      Fileset.Type type,
-      String storageLocation,
-      Map<String, String> properties)
-      throws NoSuchSchemaException, FilesetAlreadyExistsException {
-    NameIdentifier catalogIdent = getCatalogIdentifier(ident);
-    doWithCatalog(
-        catalogIdent,
-        c ->
-            c.doWithPropertiesMeta(
-                p -> {
-                  validatePropertyForCreate(p.filesetPropertiesMetadata(), properties);
-                  return null;
-                }),
-        IllegalArgumentException.class);
-    long uid = idGenerator.nextId();
-    StringIdentifier stringId = StringIdentifier.fromId(uid);
-    Map<String, String> updatedProperties =
-        StringIdentifier.newPropertiesWithId(stringId, properties);
-
-    Fileset createdFileset =
-        doWithCatalog(
-            catalogIdent,
-            c ->
-                c.doWithFilesetOps(
-                    f -> f.createFileset(ident, comment, type, storageLocation, updatedProperties)),
-            NoSuchSchemaException.class,
-            FilesetAlreadyExistsException.class);
-    return EntityCombinedFileset.of(createdFileset)
-        .withHiddenPropertiesSet(
-            getHiddenPropertyNames(
-                catalogIdent,
-                HasPropertyMetadata::filesetPropertiesMetadata,
-                createdFileset.properties()));
-  }
-
-  @Override
-  public Fileset alterFileset(NameIdentifier ident, FilesetChange... changes)
-      throws NoSuchFilesetException, IllegalArgumentException {
-    validateAlterProperties(ident, HasPropertyMetadata::filesetPropertiesMetadata, changes);
-
-    NameIdentifier catalogIdent = getCatalogIdentifier(ident);
-    Fileset alteredFileset =
-        doWithCatalog(
-            catalogIdent,
-            c -> c.doWithFilesetOps(f -> f.alterFileset(ident, changes)),
-            NoSuchFilesetException.class,
-            IllegalArgumentException.class);
-    return EntityCombinedFileset.of(alteredFileset)
-        .withHiddenPropertiesSet(
-            getHiddenPropertyNames(
-                catalogIdent,
-                HasPropertyMetadata::filesetPropertiesMetadata,
-                alteredFileset.properties()));
-  }
-
-  @Override
-  public boolean dropFileset(NameIdentifier ident) {
-    return doWithCatalog(
-        getCatalogIdentifier(ident),
-        c -> c.doWithFilesetOps(f -> f.dropFileset(ident)),
-        NonEmptyEntityException.class);
-  }
-
-  private <R, E extends Throwable> R doWithCatalog(
+  <R, E extends Throwable> R doWithCatalog(
```

Comment:
Why do you change to package private, not protected?

Reply:
The `protected` modifier provides a wider scope of visibility than `package private`, do we need `protected`?